src/cljc/athens/common_events/resolver.cljc

```diff
@@ -861,6 +861,47 @@
     tx-data))
 
 
+(defmethod resolve-event-to-tx :datascript/selected-delete
+  [db {:event/keys [args]}]
+  ;; We know that we only need to dec indices after the last block. The former blocks
+  ;; are necessarily going to remove all tail children, meaning we only need to be
+  ;; concerned with the last N blocks that are selected, adjacent siblings, to
+  ;; determine the minus-after value.
+  (println "resolver :datascript/drop-multi-child args" (pr-str args))
+  (let [{:keys [uids]}                    args
+        selected-sibs-of-last             (->> uids
+                                               (d/q '[:find ?sib-uid ?o
+                                                      :in $ ?uid [?selected ...]
+                                                      :where
+                                                      ;; get all siblings of the last block
+                                                      [?e :block/uid ?uid]
+                                                      [?p :block/children ?e]
+                                                      [?p :block/children ?sib]
+                                                      [?sib :block/uid ?sib-uid]
+                                                      ;; filter selected
+                                                      [(= ?sib-uid ?selected)]
+                                                      [?sib :block/order ?o]]
+                                                    db
+                                                    (last uids))
+                                               (sort-by second))
+        [uid order]                       (last selected-sibs-of-last)
+        parent-eid                        (:db/id (common-db/get-parent db [:block/uid uid]))
+        n                                 (count selected-sibs-of-last)
+        reindex                           (common-db/minus-after db
+                                                                 parent-eid
```

Comment:
I don't think we can select multiple blocks located anywhere and delete them. These are the ways in which block(s) can be selected:

- A single block located anywhere can be selected
- Multiple consecutive blocks, we cannot select blocks that are not consecutive

So in the case of:
- A single block being selected we reindex the blocks under this block's parent
- Multiple blocks we reindex only the last selected block's parent, because of how the selection works we can have 2 types of selected blocks:
     -  Blocks which have the same parent as the last selected block
     -  Blocks that don't have the same parent as the last selected block, in this case, we don't need to reindex these blocks parents because these blocks are always going to be tail children i.e they represent the last "X" children of their parent and removing them will not affect the order. You can see the example below 
     - ![Screenshot from 2021-07-28 13-08-56](https://user-images.githubusercontent.com/22994594/127252911-a9cac2c7-a1cb-4861-afcf-8ad86c64b60c.png)
     - In the above example we don't need to reindex parents of blocks : 8, 10 and 12 because the selected blocks under their parents are tail children
     - ![Screenshot from 2021-07-28 13-10-25](https://user-images.githubusercontent.com/22994594/127253016-a356d925-e423-4f3c-bdb4-ce50dd8f8095.png)
     - In the above example we don't need to reindex the parents of blocks: 6, 7(child of 6) and 12  because either the selected blocks under those parents tail children or the parent itself is deleted (in case of block 7)

Reply:
Ok I understand what's going on better now. Thank you for explaining it to me. 

I agree that the fact selections are contiguous enables this optimization whereby only a single parent's children needs to be computed. I didn't put a lot of stock on the event name (`:datascript/selected-delete`) but now I understand the bit about it being a selection is very relevant.

Can you add a comment to this effect somewhere in this defmethod please? Maybe something like this above the `reindex` binding:
```
;; Since the selection is always contiguous, there's only one parent (the last one) whose children need to be reindexed.
```